From 89eda41590411f9402a62c05647c4c271af40436 Mon Sep 17 00:00:00 2001
From: Peter Maydell <peter.maydell@linaro.org>
Date: Mon, 18 Feb 2013 16:58:27 +0000
Subject: [PATCH 24/70] hw/omap_i2c: Treat 32 bit accesses like 16 bit accesses

Treat 32 bit OMAP i2c register accesses like 16 bit accesses, working
around the kernel bug LP:727781 where it makes 32 bit accesses in violation
of the TRM. (Hardware behaviour is apparently that OMAP2 will abort on
these 32 bit accesses, but OMAP3 and 4 will allow them.)

We might need to revert or rethink this patch as part of handling silencing
of bad-kernel-behaviour warnings more consistently so that they can be
upstreamed.

Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
---
 hw/i2c/omap_i2c.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/i2c/omap_i2c.c b/hw/i2c/omap_i2c.c
index 93e85d4..353ca46 100644
--- a/hw/i2c/omap_i2c.c
+++ b/hw/i2c/omap_i2c.c
@@ -649,12 +649,12 @@ static const MemoryRegionOps omap_i2c_ops = {
         .read = {
             omap_i2c_readb,
             omap_i2c_read,
-            omap_badwidth_read16,
+            omap_i2c_read,
         },
         .write = {
             omap_i2c_writeb, /* Only the last fifo write can be 8 bit.  */
             omap_i2c_write,
-            omap_badwidth_write16,
+            omap_i2c_write,
         },
     },
     .endianness = DEVICE_NATIVE_ENDIAN,
-- 
1.8.5.2

